src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -297,8 +297,14 @@ IContainSagaData CreateNewSagaEntity(SagaMetadata metadata, IInvokeHandlerContex
                     .ConvertFromInvariantString(value.PropertyValue.ToString());
 
                 propertyInfo.SetValue(sagaEntity, convertedValue);
-            }
 
+                sagaEntity.Id = sagaIdGenerator.Generate(value.PropertyName, value.PropertyValue, metadata, context.Extensions);
+            }
+            else
+            {
+                sagaEntity.Id = sagaIdGenerator.Generate(null, null, metadata, context.Extensions);
```

Comment:
How about we pass in https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Sagas/SagaCorrelationProperty.cs#L6 which has `.None` (which ends up being `null `as well but anyway ;) )